src/pyasm/widget/file_wdg.py

```diff
@@ -1151,8 +1151,9 @@ def get_icon_info(my, image_link, repo_path=None, icon_type='icon'):
 
             # HACK for pdf icons
             if image_link.endswith(".pdf"):
-                if icon_size.endswith("%"):
-                    icon_size = float(icon_size[0:-1])
+                #check if icon_size is a string endswith %
+                if isinstance(icon_size, str):
+                    icon_size = float(icon_size.strip('%'))/100
```

Comment:
what if the user set icon_size as 20em, refer to the get_icon_size() method in the same file and see how it skips calculation when it's 100% or 20em..  get_icon_size() may not cover all cases either. but check that first